File: solutions/go/pangram/1/pangram.go
package pangram

import "strings"

func IsPangram(input string) bool {
	input = strings.ToLower(input)

	var alphabet = map[rune]bool{
		'a': false,
		'b': false,
		'c': false,
		'd': false,
		'e': false,
		'f': false,
		'g': false,
		'h': false,
		'i': false,
		'j': false,
		'k': false,
		'l': false,
		'm': false,
		'n': false,
		'o': false,
		'p': false,
		'q': false,
		'r': false,
		's': false,
		't': false,
		'u': false,
		'v': false,
		'w': false,
		'x': false,
		'y': false,
		'z': false,
	}
	for _, c := range input {
		if alphabet[c] == false {
			alphabet[c] = true
		}
	}

	for _, v := range alphabet {
		if v == false {
			return false
		}
	}
	return true
}

File: solutions/c/pangram/1/pangram.c
#include "pangram.h"
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

bool is_pangram(const char *sentence) {
  if (sentence == NULL) {
    return false;
  }

  size_t len = strlen(sentence);
  int seen = 0;
  char c;

  for (size_t i = 0; i < len; i++) {
    c = sentence[i];
    if (c >= 'A' && c <= 'Z') {
      c = tolower(c);
    }

    if (c >= 'a' && c <= 'z') {
      seen |= 1 << (c - 'a');
    }
  }

  return seen == ((1 << 26) - 1);
}
